### PR TITLE
Clean up some MockLogAppender usages

### PR DESCRIPTION
**Pull request by DaveCTurner**
No need for the explict start/add...remove/stop any more.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java [hunk 1]
```diff
@@ -400,10 +396,8 @@ public void testDebugLogging() {
 
             runTasksUntil(deterministicTaskQueue, CLUSTER_NODE_RECONNECT_INTERVAL_SETTING.get(Settings.EMPTY).millis());
             appender.assertAllExpectationsMatched();
-        } finally {
```
Comment (tlrx) [inline server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java:403]: Do you think we can check that `assertAllExpectationsMatched` has been called at least once before closing the captured releasable?

**Review by tlrx — APPROVED** [review]
LGTM, I left a comment

File: server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java:403]: Yes that makes sense - indeed we should have called `assertMatched` on every registered expectation. I won't be doing that here but a followup would be welcome.

Comment (tlrx) [reply server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java:403]: I can take this as a follow up, no need for you to spend time on this now.

**Review by idegtiarenko — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java (hunk 1 above)
Comment (tlrx) [reply server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java:403]: I opened #92765

**Pull request merged**